middle_end/simplify_primitives.ml

```diff
@@ -147,6 +147,7 @@ let primitive (p : Lambda.primitive) (args, approxs) expr dbg ~size_int
         | Ostype_unix -> S.const_bool_expr expr (Sys.os_type = "Unix")
         | Ostype_win32 -> S.const_bool_expr expr (Sys.os_type = "Win32")
         | Ostype_cygwin -> S.const_bool_expr expr (Sys.os_type = "Cygwin")
+        | Backend_type -> S.const_int_expr expr 0 (* Native *)
```

Comment:
Why is const_in_expr used here vs make_const_bool in closure.ml?  This might be completely harmless, but the lack of symmetry looks suspicious.